connect/client.go

```diff
@@ -14,7 +14,67 @@ func Register() error {
 		return err
 	}
 
-	// TODO remainder of Register()
+	installReleasePkg := true
+	product := CFG.Product
+	if product.isEmpty() {
+		product, err = baseProduct()
+		if err != nil {
+			return err
+		}
+		installReleasePkg = false
+	}
+
+	if err = registerProduct(product, installReleasePkg); err != nil {
+		return err
+	}
+
+	if product.IsBase {
+		p, err := showProduct(product)
+		if err != nil {
+			return err
+		}
+		if err := registerProductTree(p); err != nil {
+			return err
+		}
+	}
+	fmt.Print(bold(greenText("\nSuccessfully registered system\n")))
+	return nil
+}
+
+// registerProduct activates the product, adds the service and installs the release package
+func registerProduct(product Product, installReleasePkg bool) error {
+	fmt.Printf("\nActivating %s %s %s ...\n", product.Name, product.Version, product.Arch)
+	service, err := activateProduct(product, CFG.Email)
+	if err != nil {
+		return err
+	}
+	fmt.Println("-> Adding service to system ...")
+	// TODO no_zypper_refs
+	if err := addService(service.URL, service.Name, true); err != nil {
+		return err
+	}
+	if installReleasePkg {
+		fmt.Println("-> Installing release package ...")
+		// TODO product.Identifier ? seems to be same as Name
+		if err := installReleasePackage(product.Name); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// registerProductTree traverses (depth-first search) the product
+// tree and registers the recommended and available products
+func registerProductTree(product Product) error {
+	for _, extension := range product.toExtensionsList() {
```

Comment:
This returns flattened tree of extensions. Product.extensions should be used directly here.

Reply:
Maybe we should rename this function to make it more clear.

Reply:
> This returns flattened tree of extensions. Product.extensions should be used directly here.

ah okay

Reply:
done